src/structure/vector.rs

```diff
@@ -409,6 +415,158 @@ impl FPVector for Vec<f64> {
     }
 }
 
+impl ParallelFPVector for Vec<f64> {
+    type Scalar = f64;
+
+    /// par_fmap for `Vec<f64>`
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use]
+    /// extern crate peroxide;
+    /// use peroxide::fuga::*;
+    /// use peroxide::traits::fp::ParallelFPVector;
+    ///
+    /// fn main() {
+    ///     let a = c!(1,2,3,4,5);
+    ///     assert_eq!(a.par_fmap(|x| x*2f64), seq!(2,10,2));
+    /// }
+    /// ```
+    fn par_fmap<F>(&self, f: F) -> Vec<f64>
+    where
+        F: Fn(f64) -> f64 + Sync + Send,
+    {
+        let mut v = self.clone();
+        v.par_iter_mut().for_each(|x| *x = f(*x));
+        v
+    }
+
+    /// reduce for `Vec<f64>`
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use]
+    /// extern crate peroxide;
+    /// use peroxide::fuga::*;
+    /// use peroxide::traits::fp::ParallelFPVector;
+    ///
+    /// fn main() {
+    ///     let a = seq!(1,100,1);
+    ///     assert_eq!(a.par_reduce(0, |x,y| x + y), 5050f64);
+    /// }
+    /// ```
+    fn par_reduce<F, T>(&self, init: T, f: F) -> f64
+    where
+        F: Fn(f64, f64) -> f64 + Send + Sync,
+        T: Into<f64> + Send + Sync + Copy,
+    {
+        self.iter().fold(init.into(), |x, &y| f(x, y))
+
+        // Parallel version unimplemented
```

Comment:
I am struggling to make a parallel version of reduce, to be honest. The commented out part of the code is my attempt, but the tests fail.